retrieval/scrape.go

```diff
@@ -65,11 +65,30 @@ var (
 		},
 		[]string{interval},
 	)
+	targetScrapeFailures = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "target_scrape_failures_total",
+			Help:      "Total number of scrapes that failed due to HTTP failure.",
+		},
+		[]string{interval},
```

Comment:
Doesn't cover the reload duration, which will be interesting for debugging (at least I cannot give you an accurate estimate how it behaves under high load).